src/extension.ts

```diff
@@ -521,6 +539,53 @@ class Device extends vscode.TreeItem {
             [this.shellPort.toString()]);
         term.show();
     }
+    
+    async captureScreenshot() {
+        const statusBarMessage = new StatusBarProgressionMessage("Attempting to capture screenshot...");
+
+        const handleCaptureError = (e) => {
+            vscode.window.showErrorMessage("Error capturing screenshot: " + (e.message || e));
+            statusBarMessage.finish();
+        }
+
+        try {
+            const screenshotDirectory = await getSharedTempDir('ev3dev-screenshots');
+            const screenshotBaseName = `ev3dev-${sanitizedDateString()}.png`;
+            const screenshotFile = `${screenshotDirectory}/${screenshotBaseName}`;
+
+            const conn = await this.exec('fbgrab -');
+            const writeStream = fs.createWriteStream(screenshotFile);
+
+            conn.on('error', (e: Error) => {
+                writeStream.removeAllListeners('finish');
+                handleCaptureError(e);
+            });
+
+            writeStream.on('open', () => {
+                conn.stdout.pipe(writeStream);
+            });
+            
+            writeStream.on('error', (e: Error) => {
+                vscode.window.showErrorMessage("Error saving screenshot: " + e.message);
+                statusBarMessage.finish();
+            });
+
+            writeStream.on('finish', async () => {
+                const pngHeader = [ 0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A ];
+                if(await verifyFileHeader(screenshotFile, pngHeader)) {
+                    statusBarMessage.finish(`Screenshot "${screenshotBaseName}" successfully captured`);
+                    vscode.commands.executeCommand('vscode.open', vscode.Uri.file(screenshotFile));
+                }
+                else {
+                    handleCaptureError("The captured screenshot was not in the expected format."
+                        + " This may be caused by running an old version of 'fbcat' on the remote device; try updating the fbcat package.");

```

Comment:
Also, on my 13" macbook, this message gets cut off because it is so long. The full message is visible in a tooltip, but it is a bit hard to read.

![image](https://user-images.githubusercontent.com/963645/29675210-21e7cbf4-88ba-11e7-8459-21ec0555c870.png)


Suggestion: you could make the message much shorter and then include a "More Info..." button that links to a wiki page or displays a second info box with the details.